trio/tests/test_util.py

```diff
@@ -93,6 +95,9 @@ def not_main_thread():
 
 # @coroutine is deprecated since python 3.8, which is fine with us.
 @pytest.mark.filterwarnings("ignore:.*@coroutine.*:DeprecationWarning")
+@pytest.mark.skipif(
```

Comment:
Can you guard just the actual use of `@asyncio.coroutine` with this condition? The test covers many different errors and most of them should still work on 3.11.

Reply:
Done.